cmd/generic-handlers.go

```diff
@@ -73,6 +75,9 @@ const (
 // and must not set by clients
 func containsReservedMetadata(header http.Header) bool {
 	for key := range header {
+		if slices.Contains(maps.Keys(validReplicationHeaders), key) {
+			return false
+		}
```

Comment:
what is the performance impact here? can you measure it with `go test -bench .` ?

Reply:
sure, will do this and update